pkg/manager/member/tiflash_scaler_test.go

```diff
@@ -0,0 +1,1839 @@
+				}
+				tc.Status.TiFlash.TombstoneStores[pod.ID] = pod
+				return nil, nil
+			})
+		}
+		if test.pvcUpdateErr {
+			pvcControl.SetUpdatePVCError(errors.NewInternalError(fmt.Errorf("API server failed")), 0)
+		}
+
+		tmpNewSet := newSet.DeepCopy()
+		err := scaler.ScaleIn(tc, oldSet, tmpNewSet)
+		test.errExpectFn(g, err)
+		g.Expect(int(*tmpNewSet.Spec.Replicas)).To(Equal(test.newReplicas))
+
+		if test.extraTestFn != nil {
+			test.extraTestFn(g, tc, scaler, tmpNewSet, newSet)
+		}
+	}
+
+	tests := []testcase{
+		{
+			name:         "1 scaleInParallelism, store is up, delete store failed",
+			storeFun:     normalTiFlashStoreFun,
+			delStoreErr:  true,
+			pvcUpdateErr: false,
+			errExpectFn:  errExpectNotNil,
+			newReplicas:  5,
+			pods: []podStatus{{
+				hasPVC:        true,
+				storeIDSynced: true,
+				isPodReady:    true,
+				hasSynced:     true,
+				ordinal:       4,
+				storeIdLabel:  "1",
+			}, {
+				hasPVC:        true,
+				storeIDSynced: true,
+				isPodReady:    true,
+				hasSynced:     true,
+				ordinal:       3,
+				storeIdLabel:  "13",
+			}},
+			scaleInParallelism: 1,
+		}, {
+			name:         "1 scaleInParallelism, store is up",
+			storeFun:     normalTiFlashStoreFun,
+			delStoreErr:  false,
+			pvcUpdateErr: false,
+			errExpectFn:  errExpectRequeue,
+			newReplicas:  5,
+			pods: []podStatus{{
+				hasPVC:        true,
+				storeIDSynced: true,
+				isPodReady:    true,
+				hasSynced:     true,
+				ordinal:       4,
+				storeIdLabel:  "1",
+			}, {
+				hasPVC:        true,
+				storeIDSynced: true,
+				isPodReady:    true,
+				hasSynced:     true,
+				ordinal:       3,
+				storeIdLabel:  "13",
+			}},
+			scaleInParallelism: 1,
+		}, {
+			name:         "1 scaleInParallelism, store state is tombstone, update pvc err",
+			storeFun:     multitombstoneTiFlashStoreFun,
+			delStoreErr:  false,
+			pvcUpdateErr: true,
+			errExpectFn:  errExpectNotNil,
+			newReplicas:  5,
+			pods: []podStatus{{
+				hasPVC:        true,
+				storeIDSynced: true,
+				isPodReady:    true,
+				hasSynced:     true,
+				ordinal:       4,
+				storeIdLabel:  "1",
+			}, {
+				hasPVC:        true,
+				storeIDSynced: true,
+				isPodReady:    true,
+				hasSynced:     true,
+				ordinal:       3,
+				storeIdLabel:  "13",
+			}},
+			scaleInParallelism: 1,
+		}, {
+			name:         "1 scaleInParallelism, store state is tombstone",
+			storeFun:     multitombstoneTiFlashStoreFun,
+			delStoreErr:  false,
+			pvcUpdateErr: false,
+			errExpectFn:  errExpectNil,
+			newReplicas:  4,
+			pods: []podStatus{{
+				hasPVC:        true,
+				storeIDSynced: true,
+				isPodReady:    true,
+				hasSynced:     true,
+				ordinal:       4,
+				storeIdLabel:  "1",
+			}, {
+				hasPVC:        true,
+				storeIDSynced: true,
+				isPodReady:    true,
+				hasSynced:     true,
+				ordinal:       3,
+				storeIdLabel:  "13",
+			}},
+			scaleInParallelism: 1,
+		}, {
+			name:         "2 scaleInParallelism, store is up",
```

Comment:
do we need to exchange L1059 and LL1082?

Reply:
Yep, you are right.